app/addons/fauxton/components.react.jsx

```diff
@@ -269,12 +269,63 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  // A super-simple replacement for window.confirm()
+  var ConfirmationModal = React.createClass({
+    propTypes: {
+      visible: React.PropTypes.bool.isRequired,
+      text: React.PropTypes.string.isRequired,
+      onClose: React.PropTypes.func.isRequired,
+      onSubmit: React.PropTypes.func.isRequired
```

Comment:
As per our last conversation, I think this is okay. The parent component just fires an action in the onClose and onSubmit methods and doesn't modify itself directly.